### ISSUE TITLE
Support for newer versions of Keycloak?

### ISSUE THREAD

**Issue by Just-Insane** [mulesoft-labs/keycloak-duo-spi#6]
Just wondering if there is any support for newer versions of keycloak (or if there are issues with newer versions if issues will be investigated?).

Also looking to add this to a keycloak install inside Kubernetes, but that should be easy enough.

**Comment by joeknock90** [comment]
Also looking for this. Stuck on the last major version.

**Comment by pgiblock** [comment]
We are about to switch away from Keycloak due to this. Hard to believe Duo isn't natively supported given how popular it is, and there aren't any active or even marginally supported extensions for it.

**Comment by Just-Insane** [comment]
One option might be to use Duo’s authentication proxy between Keycloak and your identity provider (such as an LDAP server like Active Directory).

It’s not ideal and there won’t be any UI, just a push notification on the user’s phone, but it would probably work. I’ve tested similar setups in the past, but not Keycloak specifically.

**Comment by pgiblock** [comment]
True. But that would prevent use of a security key. 😢

**Comment by Just-Insane** [comment]
This is true, but I wonder if Keycloak’s webauthn features would mitigate that?

Then you use your username/password + duo, and then an additional security key on top of that.

kinda kludgy, but I’m not sure how you’d use a hardware key with Duo and Keycloak as is. Pretty sure this solution only supported prompts in the past (it’s been a while).

**Comment by lehighkid** [comment]
I was able to update to 15.0.2 using the OSC fork and docker build with one minor adjustment - I needed to move duo-mfa.ftl from the root directory to `main/resources/theme-resources/templates` and then it all worked.  I have since; however, shifted projects to leverage Duo Universal Prompt see issue #7

**Comment by treydock** [comment]
@littlechicks https://github.com/OSC/keycloak-duo-spi - The main reason for fork was to add group support so only certain groups require Duo.  I use the Makefile commands to build the jar with Docker.  We are using that SPI on Keycloak 16.1.1.  I'm not sure how well it will work with major changes in Keycloak 17 but will find out in coming months when we upgrade.
